diff --git a/source/duke3d/src/common.cpp b/source/duke3d/src/common.cpp
index db09fb5b4..006561b00 100644
--- a/source/duke3d/src/common.cpp
+++ b/source/duke3d/src/common.cpp
@@ -597,10 +597,18 @@ void G_AddSearchPaths(void)
 #if defined __linux__ || defined EDUKE32_BSD
     char buf[BMAX_PATH];
     char *homepath = Bgethomedir();
+    char *xdg_docs_path = getenv("XDG_DOCUMENTS_DIR");
 
     Bsnprintf(buf, sizeof(buf), "%s/.steam/steam", homepath);
     Duke_AddSteamPaths(buf);
 
+    // Steam Flatpak
+    Bsnprintf(buf, sizeof(buf), "%s/.var/app/com.valvesoftware.Steam/.steam/steam", homepath);
+    Duke_AddSteamPaths(buf);
+
+    Bsnprintf(buf, sizeof(buf), "%s/.var/app/com.valvesoftware.Steam/.steam/steam/steamapps/libraryfolders.vdf", homepath);
+    Paths_ParseSteamLibraryVDF(buf, Duke_AddSteamPaths);
+
     Bsnprintf(buf, sizeof(buf), "%s/.steam/steam/steamapps/libraryfolders.vdf", homepath);
     Paths_ParseSteamLibraryVDF(buf, Duke_AddSteamPaths);
 
@@ -614,12 +622,23 @@ void G_AddSearchPaths(void)
     Fury_Add_GOG_Linux(buf);
     Paths_ParseXDGDesktopFilesFromGOG(homepath, "ION_Fury", Fury_Add_GOG_Linux);
 
+    if (xdg_docs_path) {
+        Bsnprintf(buf, sizeof(buf), "%s/Eduke32", xdg_docs_path);
+        addsearchpath(buf);
+    }
+    else {
+        Bsnprintf(buf, sizeof(buf), "%s/Documents/EDuke32", homepath);
+        addsearchpath(buf);
+    }
+
     Xfree(homepath);
+    Xfree(xdg_docs_path);
 
     addsearchpath("/usr/share/games/jfduke3d");
     addsearchpath("/usr/local/share/games/jfduke3d");
     addsearchpath("/usr/share/games/eduke32");
     addsearchpath("/usr/local/share/games/eduke32");
+    addsearchpath("/app/extensions/extra");
 #elif defined EDUKE32_OSX
     char buf[BMAX_PATH];
     int32_t i;
