diff --git a/GNUmakefile b/GNUmakefile
index 54618b2e6..53ab71d0d 100644
--- a/GNUmakefile
+++ b/GNUmakefile
@@ -811,6 +811,9 @@ libklzw$(DLLSUFFIX): $(engine_src)/klzw.cpp
 	$(COMPILE_STATUS)
 	$(RECIPE_IF) $(COMPILER_C) -shared -fPIC $< -o $@ $(RECIPE_RESULT_COMPILE)
 
+%.png: %.bmp
+	ffmpeg -i $< -frames:v 1 -update 1 $@
+
 # to debug the tools link phase, make a copy of this rule explicitly replacing % with the name of a tool, such as kextract
 %$(EXESUFFIX): $(tools_obj)/%.$o $(foreach i,tools $(tools_deps),$(call expandobjs,$i))
 	$(LINK_STATUS)
@@ -893,7 +896,7 @@ $$($1_obj)/%.$$o: $$($1_rsrc)/%.c | $$($1_obj)
 	$$(call MKDIR,$$(dir $$@))
 	$$(RECIPE_IF) $$(COMPILER_C) $$($1_cflags) -c $$< -o $$@ $$(RECIPE_RESULT_COMPILE)
 
-$$($1_obj)/%_banner.c: $$($1_rsrc)/%.bmp | $$($1_obj)
+$$($1_obj)/%_banner.c: $$($1_rsrc)/%.png | $$($1_obj)
 	echo "#include \"gtkpixdata_shim.h\"" > $$@
 	gdk-pixbuf-csource --extern --struct --raw --name=startbanner_pixdata $$^ | sed 's/load_inc//' >> $$@
 
